api/types/resource.go

```diff
@@ -83,6 +84,18 @@ func GetName[R Resource](r R) string {
 	return r.GetName()
 }
 
+// ResourceNameIter creates an iterator that loops through the provided slice of
+// resources and return their names.
+func ResourceNameIter[R Resource, S ~[]R](s S) iter.Seq[string] {
```

Comment:
```suggestion
func ResourceNames[R Resource, S ~[]R](s S) iter.Seq[string] {
```

https://pkg.go.dev/iter#hdr-Naming_Conventions